cat/linux-kernel.cat

```diff
@@ -18,7 +18,7 @@
  * It can be replaced by include "cos.cat" for tests that do not use locks.
  *)
 
-include "lock.cat"
+//include "lock.cat"
```

Comment:
I am fine with modifying / cleaning some of the cat files (I am actually doing while making better use of the includes). The only model I don't want to touch is lkmm: the version in the kernel tree is updated every now and then and when I update our version I want to keep the required changes at minimum (with parametric relations the only remaining change is moving the def of Marked from the bell file to the cat one IIRC).

What is the problem with lock.cat? We should leave the include as it is and we either remove the file from our repo, or we commented out in lock.cat whatever is causing problems (the later might have the advantage of still performing some sanity checks in the form of flagged axioms).

Reply:
> What is the problem with lock.cat? 

```
let RL = try RL with emptyset
let RU = try RU with emptyset
...
let all-possible-rfe-lf =
	let possible-rfe-lf r =
		let pair-to-relation p = p ++ 0
		in map pair-to-relation ((LKW * {r}) & loc & ext)
	in map possible-rfe-lf (LF \ range(rfi-lf))
...
with rfe-lf from cross(all-possible-rfe-lf)
...
let rf = rf | ([IW | UL] ; singlestep(co) ; lk-rmw^-1)
...
show co,rf,fr
```
There is a lot of unsupported stuff. And most definitions are some very roundabout way to add lock events to `rf/co`, something which we would probably do differently.

> We should leave the include as it is and we either remove the file from our repo, or we commented out in lock.cat whatever is causing problems (the later might have the advantage of still performing some sanity checks in the form of flagged axioms).
We can also delete that file. But including the sanity checks seems like a bad idea: you pay the price for all `cat_spec` verifications for something that you should never need to verify.

Reply:
Yes, I spent some more time this morning cleaning up the models / Wmm.class and came up to the conclusion that the best is to comment the whole content of `lock.cat` and add inside it an include to `cos.cat` (which is the one where I define all basic stuff).

I am running some tests right now and hopefully will push (make use of parametric and include) soon.

Reply:
Hmm, I wouldn't abuse `cos.cat` for the basic stuff because that file is a standard one in `herd7` with the special purpose of generating coherence (and `cos.cat` is used in more models than just LKMM).
I would rather have a completely own `basic.cat` (or whatever you want to call it) and you could include that. You could then have `cos.cat` just include `basic.cat` so you don't need to change the memory model includes.

Reply:
We don't yet have a `cos.cat` file in our repo. Copying the file from herd's repo might cause problems similar to `lock.cat` (i.e. unsupported stuff). Adding an empty one that just includes `basic.cat` seems pointless.

My goal is to keep all models using `cos.cat` untouched. When we run those models with dat3m, it would include our own `cos.cat`. If one would want to use our models with herd, it would have to run something like `herd8 -model cat/X.cat -include /herdrepo/...`, thus picking up the one that herd needs

Reply:
I still think it would be cleaner to have it just include `basic.cat`. For example, in those models that herd7 does not support/have (GPU models), we would also like to include those definitions but including `cos.cat` suggests that this is somehow related to herd7 or that herd7 supports those files somehow.
Apart from that, `cos.cat` can include `basic.cat` and have a comment/explanation of what is going on here (i.e. that this is a mock that is different to herd7's version). Just having such a comment is IMO enough to justify that.

Reply:
Done